src/drivers/telemetry/iridiumsbd/IridiumSBD.h

```diff
@@ -167,114 +146,194 @@ class IridiumSBD : public device::CDev
+	void write_at(const char *command);
 
 	/*
-	 *
+	 * Read return from modem and store it in rx_command_buf
 	 */
-	satcom_result_code read_at(uint8_t *rx_buf, int *rx_len);
+	satcom_result_code read_at_command(int16_t timeout = 100);
 
 	/*
-	 *
+	 * Read return from modem and store it in rx_msg_buf
 	 */
-	void schedule_test(void);
+	satcom_result_code read_at_msg(int16_t timeout = 100);
 
 	/*
-	 *
+	 * Read the return from the modem
 	 */
-	void standby_loop(void);
+	satcom_result_code read_at(uint8_t *rx_buf, int *rx_len, int16_t timeout = 100);
 
 	/*
-	 *
+	 * Schedule a test (set test_pending to true)
 	 */
-	void csq_loop(void);
+	void schedule_test(void);
 
 	/*
-	 *
+	 * Clear the MO message buffer
 	 */
-	void sbdsession_loop(void);
+	bool clear_mo_buffer();
 
 	/*
-	 *
+	 * Open and configure the given UART port
 	 */
-	void test_loop(void);
+	satcom_uart_status open_uart(char *uart_name);
 
 	/*
-	 * TEST
+	 * Checks if the modem responds to the "AT" command
 	 */
-	void start_test(void);
+	bool is_modem_ready(void);
 
 	/*
-	 *
+	 * Get the poll state
 	 */
-	void start_sbd_session(void);
+	pollevent_t poll_state(struct file *filp);
 
 	/*
-	 * Checks if the modem responds to the "AT" command
+	 * Publish the up to date telemetry status
 	 */
-	bool is_modem_ready(void);
+	void publish_telemetry_status(void);
 
-	/*
-	 * Send a AT command to the modem
+	/**
+	 * Notification of the first open of CDev.
+	 *
+	 * This function is called when the device open count transitions from zero
+	 * to one.  The driver lock is held for the duration of the call.
+	 *
+	 * Notes that CDev is used and blocks stopping the driver.
+	 *
+	 * @param filep		Pointer to the NuttX file structure.
+	 * @return		OK if the open should proceed, -errno otherwise.
 	 */
-	void write_at(const char *command);
+	virtual int	open_first(struct file *filep) override;
+
+	/**
+	 * Notification of the last close of CDev.
+	 *
+	 * This function is called when the device open count transitions from
+	 * one to zero.  The driver lock is held for the duration of the call.
+	 *
+	 * Notes that CDev is not used anymore and allows stopping the driver.
+	 *
+	 * @param filep		Pointer to the NuttX file structure.
+	 * @return		OK if the open should return OK, -errno otherwise.
+	 */
+	virtual int	close_last(struct file *filep) override;
+
+	static IridiumSBD *instance;
+	static int task_handle;
+	bool _task_should_exit = false;
+	int uart_fd = -1;
+
+	int32_t _param_read_interval_s = -1;
+	int32_t _param_session_timeout_s = -1;
+	int32_t _param_stacking_time_ms = -1;
+
+	hrt_abstime _last_signal_check = 0;
+	uint8_t _signal_quality = 0;
+	uint16_t _failed_sbd_sessions = 0;
+
+	bool _writing_mavlink_packet = false;
+	uint16_t _packet_length = 0;
+
+	orb_advert_t _telemetry_status_pub = nullptr;
+
+	bool _test_pending = false;
+	char _test_command[32];
+	hrt_abstime _test_timer = 0;
+
+	uint8_t _rx_command_buf[SATCOM_RX_COMMAND_BUF_LEN] = {0};
```

Comment:
I suggest you change it to `= {}`. `{0}` is confusing, as for example using `{1}` would lead to only the first element being 1 and all others 0.

Reply:
Changed